### DISCUSSION TITLE
What is the best way to disable a package manager?

### DISCUSSION THREAD

**Discussion by tomkerkhove** [renovatebot/renovate#16196] [Request Help] [answered]
Whats the best way to disable a package manager?

I'm looking to disable NuGet, NPM & Helm-values so that we only start with Dockerfile and did the following:
```json
{
  "nuget": {
    "enabled": false
  },
  "packageRules": [
    {
      "datasources": ["docker"],
      "packagePatterns": ["^mcr.microsoft.com/dotnet/"],
      "groupName": ".NET Core Docker containers"
    }
  ]
}
```

However, I was not able to find a setting for Helm values nor for npm - Is there a way to disable it?

Also, the PR still lists all .NET projects that it found to update for .NET but I presume that's just FYI and it won't open PRs?

**Comment by rarkins** [comment]
I would have expected `nuget.enabled=false` to have stopped even extracting. In theory setting that for any manager name should disable the manager.

**Reply by tomkerkhove** [reply]
It is still showing up in the onboarding PR:
![image](https://user-images.githubusercontent.com/4345663/175225194-72830126-7fed-4b7f-916a-82445ec70c0d.png)

But if this is just FYI and it won't open PRs then that's OK.

How about the other package managers? How do I disable them because I couldn't find this in the docs.

**Reply by rarkins** [reply]
Example:

```json
{
  "npm": {
    "enabled": false
  }
}
```

Please create a reproduction repo on github.com if you need us to look any further. It doesn't sound like things are working as expected if you are seeing them listed in the onboarding PR

**Reply by tomkerkhove** [reply]
> Please create a reproduction repo on github.com if you need us to look any further. It doesn't sound like things are working as expected if you are seeing them listed in the onboarding PR

So I tried this in a GitHub repo and there it worked fine. Our scenario is Azure DevOps where it might not update the PR  or only when the pipeline runs. Is that a known scenario or should I still file a bug?

**Comment by viceice** [comment]
use a package rule to disable all first then use a second rule to enable the required manager

```json
"packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "enabled": false
    },
    {
      "matchManagers": ["dockerfile"],
      "enabled": true
    },
    {
      "datasources": ["docker"],
      "packagePatterns": ["^mcr.microsoft.com/dotnet/"],
      "groupName": ".NET Core Docker containers"
    }
  ]
```

**Reply by tomkerkhove** [reply]
That's fine but for somebody new, that's not really helping as I'm not sure how to do that with all the configuration options available 😅

**Reply by viceice** [reply]
Updated my comment 😉

**Reply by tomkerkhove** [reply]
Oh that's useful - Thanks!

**Comment by tomkerkhove** [comment, marked as answer]
It looks like `"enabledManagers": ["dockerfile"]` is what I was looking for:
https://docs.renovatebot.com/modules/manager/#limiting-enabled-managers

**Reply by viceice** [reply]
thats another solution 😉
